### DISCUSSION TITLE
Is there a rollback path? From v2.0.1 to v0.41.2

### DISCUSSION THREAD

**Discussion by angelbarrera92** [fluxcd/flux2#4102] [Q&A] [answered]
Hi!

I have a few projects where flux is managing flux itself. I test every flux upgrade and it's rollback. It was working pretty fine on versions v0.X.Y. But... I'm unable to roll back a flux project from v2.0.1 to v0.41.2.

The procedure I use to follow is as simple as:
- In a project running flux v0.X.Y, create a commit with the new gotk-components. *(`flux install --export > bootstrap/flux.yaml`)*
- Flux reconciles itself by self-updating.
- Revert the last commit. Wait for reconciliation.
  - This was working perfectly until I hit v2.0.1. I'll say it's expected, as it is a major upgrade with changes in its CRDs. But.... Is there a safe way to roll back?

Thanks!

**Comment by makkes** [comment]
> This was working perfectly until I hit v2.0.1.

What exactly is not working? What are the errors you're seeing?

**Reply by angelbarrera92** [reply]
It is unable to rollback the CRDs I guess:

```shell
$ kubectl get kustomization -n flux-system init
flux-system   init        9m55s   False   CustomResourceDefinition/gitrepositories.source.toolkit.fluxcd.io dry-run failed, reason: Invalid: CustomResourceDefinition.apiextensions.k8s.io "gitrepositories.source.toolkit.fluxcd.io" is invalid: status.storedVersions[1]: Invalid value: "v1": must appear in spec.versions
```

Also...

```shell
$ kubectl describe kustomization -n flux-system init
......
....
...
Events:
  Warning  ArtifactFailed           4m37s                kustomize-controller  failed to verify archive: computed checksum '7070d9d4XXXX780adb3a9f8d0924df4b43XXXXXXX7667bde0f0e' doesn't match provided '' (check whether file size exceeds max download size)
  Warning  ReconciliationFailed     19s (x4 over 2m23s)  kustomize-controller  CustomResourceDefinition/gitrepositories.source.toolkit.fluxcd.io dry-run failed, reason: Invalid: CustomResourceDefinition.apiextensions.k8s.io "gitrepositories.source.toolkit.fluxcd.io" is invalid: status.storedVersions[1]: Invalid value: "v1": must appear in spec.versions
.
```

**Reply by dacox** [reply]
@angelbarrera92 @makkes hey, I'm suddenly running into `kustomize-controller  failed to verify archive: computed checksum` - but I cant find a reference to max archive size anywhere.

How did you resolve? My GitRepository archive is only 450k..

**Comment by angelbarrera92** [comment]
The only way I was able to make it work is to uninstall v2 and reinstall v0.41.2 as it was described here: https://fluxcd.io/flux/installation/#uninstall

But, I would like to make it in a gitops way :)

**Reply by stefanprodan** [reply, marked as answer]
> But, I would like to make it in a gitops way :)

Not possible as Flux can't rollback its own CRDs, once you upgrade to v2.0 there is no way back, unless you reinstall.
